launchlib/memory.go

```diff
@@ -26,26 +25,36 @@ import (
 )
 
 const (
-	memGroupName = "memory"
-	memLimitName = "memory.limit_in_bytes"
+	memGroupName         = "memory"
+	cgroupV1MemLimitName = "memory.limit_in_bytes"
+	cgroupV2MemLimitName = "memory.max"
 )
 
 type MemoryLimit interface {
 	MemoryLimitInBytes() (uint64, error)
 }
 
-var DefaultMemoryLimit = NewCGroupMemoryLimit(os.DirFS("/"))
-
 type CGroupMemoryLimit struct {
 	pather CGroupPather
 	fs     fs.FS
 }
 
-func NewCGroupMemoryLimit(filesystem fs.FS) MemoryLimit {
+func NewCGroupMemoryLimit(filesystem fs.FS) (MemoryLimit, error) {
+	isCGroupV2, err := IsCGroupV2(filesystem)
+	if err != nil {
+		return CGroupMemoryLimit{}, errors.Wrap(err, "failed to determine cgroup version")
```

Comment:
I see in other places, when there's an error we're propagating, we usually use `nil` as the return type. Thoughts on doing that here?

Don't have much go experience. Your call.